src/overlay005/encounter_effect.c

```diff
@@ -0,0 +1,1673 @@
+
+    encEffect->hBlankFlag = FALSE;
+    screenSplitEfx->hBlankSystem = encEffect->fieldSystem->unk_04->hBlankSystem;
+    screenSplitEfx->splitHeight = 96;
+    screenSplitEfx->state = SCREENSPLIT_STATE_INTERPOLATE;
+    screenSplitEfx->done = &encEffect->hBlankFlag;
+
+    QuadraticInterpolationTaskFX32_Init(&screenSplitEfx->xInterpolationTask, 0, (255 * FX32_ONE), initialSpeedX, numSteps);
+    QuadraticInterpolationTaskFX32_Init(&screenSplitEfx->yInterpolationTask, 0, (96 * FX32_ONE), initialSpeedY, numSteps);
+
+    G2_SetWnd0Position(0, 0, 255, 192);
+    G2_SetWnd1Position(0, 0, 255, 192);
+
+    CoreSys_ExecuteAfterVBlank(ScreenSplitEffect_SetupTasks, screenSplitEfx, 1024);
+}
+
+static void ScreenSplitEffect_SetupTasks(SysTask *task, void *param)
+{
+    ScreenSplitEffect *screenSplitEfx = param;
+
+    screenSplitEfx->vBlankTask = CoreSys_ExecuteAfterVBlank(ScreenSplitEffect_VBlankCallback, screenSplitEfx, 1024);
+    screenSplitEfx->hBlankTask = HBlankSystem_StartTask(screenSplitEfx->hBlankSystem, ScreenSplitEffect_HBlankCallback, screenSplitEfx);
+
+    SysTask_Done(task);
+}
+
+static void ScreenSplitEffect_VBlankCallback(SysTask *task, void *param)
+{
+    ScreenSplitEffect *screenSplitEfx = param;
+
+    switch (screenSplitEfx->state) {
+    case SCREENSPLIT_STATE_INTERPOLATE:
+        QuadraticInterpolationTaskFX32_Update(&screenSplitEfx->yInterpolationTask);
+
+        if (QuadraticInterpolationTaskFX32_Update(&screenSplitEfx->xInterpolationTask)) {
+            screenSplitEfx->state++;
+        }
+
+        s32 currentX = screenSplitEfx->xInterpolationTask.currentValue >> FX32_SHIFT;
+        s32 currentY = screenSplitEfx->yInterpolationTask.currentValue >> FX32_SHIFT;
+        G2_SetWnd0Position(0, 0, 255 - currentX, 96 - currentY);
+        G2_SetWnd1Position(currentX, 96 + currentY, 255, 192);
+        break;
+    case SCREENSPLIT_STATE_FINISH:
+        ScreenSplitEffect_Finish(param);
+        break;
+    }
+}
+
+static void ScreenSplitEffect_HBlankCallback(HBlankTask *task, void *param)
+{
+    ScreenSplitEffect *screenSplitEfx = param;
+    int vCount;
+    int v2;
+    int v3;
+
+    vCount = GX_GetVCount();
+
+    if (vCount <= screenSplitEfx->splitHeight) {
+        GX_SetVisibleWnd(GX_WNDMASK_W0);
+    } else {
+        GX_SetVisibleWnd(GX_WNDMASK_W1);
+    }
+}
+
+static void ScreenSplitEffect_Finish(ScreenSplitEffect *screenSplitEfx)
+{
+    G2_SetWnd0InsidePlane(GX_WND_PLANEMASK_BG0 | GX_WND_PLANEMASK_BG1 | GX_WND_PLANEMASK_BG2 | GX_WND_PLANEMASK_BG3 | GX_WND_PLANEMASK_OBJ, TRUE);
+    G2_SetWndOutsidePlane(GX_WND_PLANEMASK_NONE, FALSE);
+    G2_SetWnd0Position(0, 0, 0, 0);
+    GX_SetVisibleWnd(GX_WNDMASK_W0);
+
+    *(screenSplitEfx->done) = TRUE;
+
+    HBlankTask_Delete(screenSplitEfx->hBlankTask);
+    screenSplitEfx->hBlankTask = NULL;
+    SysTask_Done(screenSplitEfx->vBlankTask);
+    screenSplitEfx->vBlankTask = NULL;
+}
+
+void include_ov5_021DDBE8_rodata_funcptr(void)
```

Comment:
This is a very strange construction; is there a way to simplify this somehow? This effectively looks like a no-op function to me.

Reply:
Removing is not an option because it doesn't match without them. They aren't used anywhere though. Removing anything from inside the functions also leads to a mismatch. I'll just rename them for now